src/native/corehost/hostmisc/pal.unix.cpp

```diff
@@ -352,7 +352,30 @@ bool get_extraction_base_parent_directory(pal::string_t& directory)
     }
     else
     {
-        trace::error(_X("Failed to determine default extraction location. Environment variable '$HOME' is not defined."));
+        // fallback to the POSIX standard getpwuid() library function
+        struct passwd* pwuid = NULL;
+        errno = 0;
+        do
+        {
+            pwuid = getpwuid(getuid());
+        } while (pwuid == NULL && errno == EINTR);
+
+        if (pwuid != NULL)
+        {
+            directory.assign(pwuid->pw_dir);
+            if (is_read_write_able_directory(directory))
+            {
+                return true;
+            }
+            else
+            {
+                trace::error(_X("Failed to determine default extraction location. Environment variable '$HOME' is not defined and directory reported by getpwuid(3) [%s] either doesn't exist or is not accessible for read/write."), pwuid->pw_dir);
```

Comment:
A nit - does it have any value to mention getpwuid(3) instead of just getpwuid?

Reply:
`man man` notes:
```
MANUAL SECTIONS
       The standard sections of the manual include:

       1      User Commands
       2      System Calls
       3      C Library Functions
       4      Devices and Special Files
       5      File Formats and Conventions
       6      Games et. Al.
       7      Miscellanea
       8      System Administration tools and Deamons
```
`getpwuid(3)` indicates to the user that it is a system C library function (rather than something of off .NET codebase). I can delete `3` if that's preferred. :)

Reply:
I would prefer not to have it there.